improver/lightning.py

```diff
@@ -0,0 +1,189 @@
+        )(precip)
+
+        data = cape_true.data * precip_true.data
+
+        cube = create_new_diagnostic_cube(
+            name="probability_of_number_of_lightning_flashes_per_unit_area_in_vicinity"
+            "_above_threshold",
+            units="1",
+            template_cube=precip,
+            data=data.astype(FLOAT_DTYPE),
+            mandatory_attributes=generate_mandatory_attributes(
+                cubes, model_id_attr=model_id_attr
+            ),
+        )
+
+        coord = DimCoord(np.array([0], dtype=FLOAT_DTYPE), units="s-1",)
+        coord.rename("number_of_lightning_flashes_per_unit_area")
+        coord.var_name = "threshold"
+        coord.attributes.update({"spp__relative_to_threshold": "greater_than"})
+        cube.add_aux_coord(coord)
+
+        return cube
+
+    @staticmethod
+    def _get_inputs(cubes: CubeList) -> Tuple[Cube, Cube]:
+        """
+        Separates CAPE and precipitation rate cubes and checks that the following
+        match: forecast_reference_time, spatial coords, time-bound interval and
+        that CAPE time is exactly one hour older than precipitation rate time.
+        """
+        cape = cubes.extract(
+            iris.Constraint(
+                cube_func=lambda cube: "atmosphere_convective_available_potential_energy"
+                in cube.name()
+            )
+        )
+        if cape:
+            cape = cape.merge_cube()
+        else:
+            raise ValueError(
+                f"No cube named atmosphere_convective_available_potential_energy found "
+                f"in {cubes}"
+            )
+        precip = cubes.extract(
+            iris.Constraint(
+                cube_func=lambda cube: "precipitation_rate_max" in cube.name()
+            )
+        )
+        if precip:
+            precip = precip.merge_cube()
+        else:
+            raise ValueError(f"No cube named precipitation_rate_max found in {cubes}")
+        (cape_time,) = list(cape.coord("time").cells())
+        (precip_time,) = list(precip.coord("time").cells())
+        if cape_time.point + timedelta(hours=1) != precip_time.point:
+            raise ValueError(
+                f"CAPE cube time ({cape_time.point}) should be valid one hour earlier "
+                f"than precip cube time ({precip_time.point})."
+            )
+        if np.diff(precip_time.bound) != timedelta(hours=1):
+            raise ValueError(
+                f"Precip cube time window must be one hour, not {np.diff(precip_time.bound)}."
+            )
+        if cape.coord("forecast_reference_time") != precip.coord(
+            "forecast_reference_time"
+        ):
+            raise ValueError(
+                "Supplied cubes must have the same forecast reference times"
+            )
+        if not spatial_coords_match(cape, precip):
+            raise ValueError("Supplied cubes do not have the same spatial coordinates")
+        return cape, precip
+
+
+def latitude_to_threshold(
```

Comment:
I chose to make it standalone because I'm using it to test the `LatitudeThreshold` class. I think it is better slightly separated, but I can add it to the class if you think that would be better.